src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -6355,28 +6363,35 @@ namespace SimAirServingZones {
-					FractionOfAutosize = FinalSysSizing( AirLoopNum ).ScaledHeatingCapacity;
-				}
-				if ( FinalSysSizing( AirLoopNum ).HeatOAOption == MinOA ) {
-					if ( FinalSysSizing( AirLoopNum ).DesHeatVolFlow > 0.0 ) {
-						OutAirFrac = FinalSysSizing( AirLoopNum ).DesOutAirVolFlow / FinalSysSizing( AirLoopNum ).DesHeatVolFlow;
+				if ( FinalSysSizing( AirLoopNum ).HeatingCapMethod == FractionOfAutosizedHeatingCapacity || FinalSysSizing( AirLoopNum ).ScaledHeatingCapacity == DataSizing::AutoSize ) {
+					// FinalSysSizing( AirLoopNum ).ScaledHeatingCapacity = AutoSize represents autosized heating capacity				
+					if ( FinalSysSizing( AirLoopNum ).ScaledHeatingCapacity != DataSizing::AutoSize ) FractionOfAutosize = FinalSysSizing( AirLoopNum ).ScaledHeatingCapacity;
+					if ( FinalSysSizing( AirLoopNum ).HeatOAOption == MinOA ) {
+						if ( FinalSysSizing( AirLoopNum ).DesHeatVolFlow > 0.0 ) {
+							OutAirFrac = FinalSysSizing( AirLoopNum ).DesOutAirVolFlow / FinalSysSizing( AirLoopNum ).DesHeatVolFlow;
+						} else {
+							OutAirFrac = 1.0;
+						}
+						OutAirFrac = std::min( 1.0, std::max( 0.0, OutAirFrac ) );
 					} else {
 						OutAirFrac = 1.0;
 					}
-					OutAirFrac = std::min( 1.0, std::max( 0.0, OutAirFrac ) );
-				} else {
-					OutAirFrac = 1.0;
-				}
-				if ( CurOASysNum == 0 && PrimaryAirSystem( AirLoopNum ).NumOAHeatCoils > 0 ) {
-					CoilInTemp = OutAirFrac * FinalSysSizing( AirLoopNum ).PreheatTemp + ( 1.0 - OutAirFrac ) * FinalSysSizing( AirLoopNum ).HeatRetTemp;
+					if ( CurOASysNum == 0 && PrimaryAirSystem( AirLoopNum ).NumOAHeatCoils > 0 ) {
+						CoilInTemp = OutAirFrac * FinalSysSizing( AirLoopNum ).PreheatTemp + ( 1.0 - OutAirFrac ) * FinalSysSizing( AirLoopNum ).HeatRetTemp;
+					} else {
+						CoilInTemp = OutAirFrac * FinalSysSizing( AirLoopNum ).HeatOutTemp + ( 1.0 - OutAirFrac ) * FinalSysSizing( AirLoopNum ).HeatRetTemp;
+					}
+					CoilOutTemp = FinalSysSizing( AirLoopNum ).HeatSupTemp;
+					CpAirStd = PsyCpAirFnWTdb( constant_zero, constant_twenty );
+					AutosizedCapacity = StdRhoAir * FinalSysSizing( AirLoopNum ).DesHeatVolFlow * CpAirStd * ( CoilOutTemp - CoilInTemp );
+					TempSize = FinalSysSizing( AirLoopNum ).FlowPerHeatingCapacity * AutosizedCapacity * FractionOfAutosize;
+				} else if ( FinalSysSizing( AirLoopNum ).HeatingCapMethod == HeatingDesignCapacity && FinalSysSizing( AirLoopNum ).ScaledHeatingCapacity > 0 ) {
+					TempSize = FinalSysSizing( AirLoopNum ).FlowPerHeatingCapacity * FinalSysSizing( AirLoopNum ).ScaledHeatingCapacity;
+				} else if ( FinalSysSizing( AirLoopNum ).HeatingCapMethod == CapacityPerFloorArea ) {
+					TempSize = FinalSysSizing( AirLoopNum ).FlowPerHeatingCapacity * FinalSysSizing( AirLoopNum ).ScaledHeatingCapacity * FinalSysSizing( AirLoopNum ).FloorAreaOnAirLoopCooled;
 				} else {
-					CoilInTemp = OutAirFrac * FinalSysSizing( AirLoopNum ).HeatOutTemp + ( 1.0 - OutAirFrac ) * FinalSysSizing( AirLoopNum ).HeatRetTemp;
+					// incorrect input
```

Comment:
Is it incorrect input? or should nothing be done? I assume GetInput will catch incorrect in;puts. So if the input is not what you check above then there is nothing else to do (except that these tests are && with other tests). So is everything caught here? In other words, will the else ever be executed? And what if it is?

    A10, \field Heating Design Capacity Method
      \type choice
      \key None
      \key HeatingDesignCapacity
      \key CapacityPerFloorArea
      \key FractionOfAutosizedHeatingCapacity